tests/phpunit/tests/ajax/TagSearch.php

```diff
@@ -156,4 +156,38 @@ public function test_unprivileged_user() {
 		$this->_handleAjax( 'ajax-tag-search' );
 	}
 
+	/**
+	 * Test the after_tag_search filter
+	 *
+	 * @ticket 55606
+	 * @covers ::wp_ajax_ajax_tag_search
+	 */
+	public function test_after_tag_search_filter() {
+
+		// Become an administrator.
+		$this->_setRole( 'administrator' );
+
+		// Set up a default request.
+		$_GET['tax'] = 'post_tag';
+		$_GET['q']   = 'chat';
+
+		// Add the after_tag_search filter.
+		add_filter(
+			'wp_after_tag_search',
+			static function( $results, $tax, $s ) {
+				return array( 'wp_after_tag_search was applied' );
+			}
+		);
+
+		// Make the request.
+		try {
+			$this->_handleAjax( 'ajax-tag-search', $_GET['tax'], $_GET['q'] );
+		} catch ( WPAjaxDieContinueException $e ) {
+			unset( $e );
+		}
+
+		// Ensure we found the right match.
+		$this->assertSame( $this->_last_response, 'chattels' );
```

Comment:
```suggestion
		$this->assertSame( 'wp_after_tag_search was applied', $this->_last_response );
```
- Changed the `expected` value to match the value passed in the filter callback.
- Swapped the order of the `expected` and `actual` values. (This is incorrect in the existing tests, and I'll be following up on this in a separate ticket to get these corrected).